File: src/ProjectR/content/scripts/spells/TwilightBarrier.lua
function GetTargetType()
    return Allies
end

function GetName()
  return "Twilight Barrier"
end

function GetDescription()
  return "Summon a twilight barrier to aid your allies.\nIncreases DEF & MR."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .88
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  d:BuffStat(DEF, .5)
  d:BuffStat(MR, .5)
end

File: src/ProjectR.Scripting/content/scripts/spells/TwilightBarrier.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class TwilightBarrier : SpellScriptBase
    {
        public override string Name { get { return "Twilight Barrier"; } }
        public override string Description { get { return "Summon a twilight barrier to aid your allies.\nIncreases DEF & MR."; } }

        public override TargetType TargetType { get { return TargetType.Allies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 88; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            BuffStat(Stat.DEF, .5);
            BuffStat(Stat.MR, .5);
        }
    }
}